draft-ietf-ppm-dap.md

```diff
@@ -751,7 +751,7 @@ To get the aggregate of a batch, the Collector issues a query specifying the
 batch ID of interest (see {{query}}). The Collector may not know which batch ID
 it is interested in; in this case, it can also issue a query of type
 `current_batch`, which allows the Leader to select a recent batch to aggregate.
-The Leader SHOULD select a batch which has not yet began collection.
+The Leader MUST select a batch that has not yet began collection.
```

Comment:
Should this be "begun" instead of "began"?

More substantialy: are we being sufficiently clear about what it means for a batch to have begun collection? I think that once the leader has successfully handled `PUT /tasks/{task-id}/collection_jobs/{collection-job-id}` (regardless of whether the collector ever saw the response), the batch has "beg[u]n collection", but I'm not sure if we say that. Maybe it'd be better to say "a batch for which a collection job exists", because I think that's easier to understand in terms of the relation between a batch and a collection job resource?

Reply:
I think you're right that it's "begun" rather than "began".

I updated the text to clarify that the selected batch "has not yet been associated with a collection job", which I think clarifies any ambiguity. I think that an aggregator implementation can choose when to associate a given `current-batch` collection request to a specific batch -- e.g. it might immediately do so as part of the `PUT` request handling, or it might choose to do so later -- but this text handles both of those choices.

On a side note: I think it's worth calling out that this leaves open whether we need to care about aggregation parameters here. That is, should the semantics be "the aggregator must select a batch which has not yet begun collection _for any aggregation parameter_" (which are the semantics we have as-written) or "not yet begun collection for the aggregation parameter in the collection request". IMO: I'm OK with going with any aggregation parameter, as this works just fine for all existing uses of DAP & the other choice is harder to implement and harder to understand.

Reply:
I think it's got to be "for any aggregation parameter", because otherwise that allows a leader to return the same batch forever.